### ISSUE TITLE
Add more model extraction attacks 

### ISSUE THREAD

**Issue by suhacker1** [trailofbits/PrivacyRaven#68]
**Is your feature request related to a problem? Please describe.**
We want every model extraction attack to be achievable in PrivacyRaven. This does not include side channel, white-box, full or partial prediction, or explanation-based attacks. 

**Describe the solution you'd like.**
PrivacyRaven has three interfaces for attacks:
 1. The core interface defines each attack parameter individually.
 2. The specific interface runs a predefined attack configuration.
 3. The cohesive interface runs every possible attack.

A user should be able to run the attack in every interface; this means that all the building blocks for the attack should be contained within PrivacyRaven. For example, new synthesizers or subset selection strategies for a specific attack should be added, so that it can be applied using the core interface. 

If you would like to implement an attack, comment with the name of the paper. Then, create a new issue referencing this issue with the name of the paper in the title. 

**Detail any additional context.**
This is a list of papers describing model extraction attacks that should be added to PrivacyRaven.
1. [**Knockoff nets: Stealing functionality of black-box models**](http://openaccess.thecvf.com/content_CVPR_2019/papers/Orekondy_Knockoff_Nets_Stealing_Functionality_of_Black-Box_Models_CVPR_2019_paper.pdf): Blocked on #10
2. [**PRADA: protecting against DNN model stealing attacks**](https://ieeexplore.ieee.org/document/8806737): Missing synthesizer
3. [**CloudLeak: Large-Scale Deep Learning Models Stealing Through Adversarial Examples**](https://www.ndss-symposium.org/ndss-paper/cloudleak-large-scale-deep-learning-models-stealing-through-adversarial-examples/): Missing some synthesizers
4. [**ACTIVETHIEF: Model Extraction Using Active Learning and Unannotated Public Data**](https://www.aaai.org/ojs/index.php/AAAI/article/view/5432): Blocked on #10
